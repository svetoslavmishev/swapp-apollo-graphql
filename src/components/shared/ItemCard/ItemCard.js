import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, CardContent, Typography } from '@material-ui/core';
import { ThemeContext } from '../../../themeContext';
import styles from './ItemCardStyles';

const ItemCard = ({
  name,
  image,
  height,
  mass,
  species,
  homeworld,
  starshipClass,
  cost,
  crew,
  maxAtmosphericSpeed,
  hyperdriveRating
}) => {
  const { currentTheme } = useContext(ThemeContext);
  const classes = styles({ currentTheme });

  return (
    <Card className={classes.card}>
      <CardContent>
        <Typography
          className={`text-center ${classes.title}`}
          variant="h6"
          component="h6"
        >
          {name}
        </Typography>
        <CardMedia className={classes.media} image={image} title={name} />

        {/* Character data */}
        {height && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Height: </span>
            <span>{height}</span>
          </Typography>
        )}
        {mass && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Weight: </span>
            <span> {mass}</span>
          </Typography>
        )}
        {species && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Species: </span>
            <span>{species.name}</span>
          </Typography>
        )}
        {homeworld && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Home World: </span>
            <span>{homeworld.name}</span>
          </Typography>
        )}

        {/* Starship data */}
        {starshipClass && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Class: </span>
            <span>{starshipClass}</span>
          </Typography>
        )}
        {cost && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Cost: </span>
            <span>{cost} credits</span>
          </Typography>
        )}
        {crew && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Crew: </span>
            <span>{crew}</span>
          </Typography>
        )}
        {maxAtmosphericSpeed && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Max Atmospheric Speed: </span>
            <span>{maxAtmosphericSpeed}</span>
          </Typography>
        )}
        {hyperdriveRating && (
          <Typography
            className={classes.descriptionDetails}
            variant="body2"
            component="p"
          >
            <span>Hyperdrive Rating: </span>
            <span>{hyperdriveRating}</span>
          </Typography>
        )}
      </CardContent>
    </Card>
  );
};

ItemCard.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  height: PropTypes.number,
  mass: PropTypes.number,
  species: PropTypes.shape({
    name: PropTypes.string
  }),
  homeworld: PropTypes.shape({
    name: PropTypes.string
  }),
  starshipClass: PropTypes.string,
  cost: PropTypes.number,
  crew: PropTypes.number,
  maxAtmosphericSpeed: PropTypes.number,
  hyperdriveRating: PropTypes.number
};

export default ItemCard;
