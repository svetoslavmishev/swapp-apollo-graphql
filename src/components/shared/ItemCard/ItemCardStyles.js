import { makeStyles } from '@material-ui/styles';

const styles = makeStyles(theme => ({
  root: {
    minHeight: 'calc(100vh - 64px)',
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].backgroundColor
  },
  card: {
    minWidth: 240,
    maxWidth: 280,
    maxHeight: 450,
    margin: 10,
    background: ({ currentTheme }) =>
      theme.palette[currentTheme].Cards.backgroundColor,
    color: ({ currentTheme }) => theme.palette[currentTheme].Cards.fontColor,
    border: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`
  },
  title: {
    fontFamily: 'StarWars',
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor
  },
  media: {
    height: 0,
    paddingTop: '100%',
    margin: '10px 0'
  },
  shipsTitle: {
    fontFamily: 'StarWars',
    textAlign: 'center',
    fontSize: 20,
    margin: 10,
    color: ({ currentTheme }) =>
      theme.palette[currentTheme].PrimaryHeading.fontColor,
    borderBottom: ({ currentTheme }) =>
      `1px solid ${theme.palette[currentTheme].Cards.borderColor}`
  },
  descriptionDetails: {
    '& span:nth-child(even)': {
      color: theme.palette.commonColors.lightBlue
    },
    '& span:nth-child(odd)': {
      color: ({ currentTheme }) =>
        theme.palette[currentTheme].PrimaryHeading.fontColor
    }
  }
}));

export default styles;
