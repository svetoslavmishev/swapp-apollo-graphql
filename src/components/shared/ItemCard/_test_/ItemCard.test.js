import React from 'react';
import { shallow } from 'enzyme';
import { ThemeContext, theme } from '../../../../themeContext';
import ItemCard from '../ItemCard';

describe('<ItemCard />', () => {
  it('renders <ItemCard /> without error', () => {
    const wrapper = shallow(
      <ThemeContext.Provider value={theme}>
        <ItemCard />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
