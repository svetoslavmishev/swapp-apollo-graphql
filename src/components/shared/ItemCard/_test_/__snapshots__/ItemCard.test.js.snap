// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemCard /> renders <ItemCard /> without error 1`] = `ShallowWrapper {}`;
